extensions/schema-compare/src/test/schemaCompareDialog.test.ts

```diff
@@ -66,4 +73,28 @@ describe('SchemaCompareDialog.openDialog @DacFx@', function (): void {
 			applyButtonState: false
 		});
 	});
+
+	it('Verify database dropdown gets populated appropriately', async function (): Promise<void> {
+		const getConnectionsResults: azdata.connection.ConnectionProfile[] = [{ ...mockConnectionProfile }];
+		sinon.stub(azdata.connection, 'getCurrentConnection').resolves(undefined);
+		sinon.stub(azdata.connection, 'openConnectionDialog').resolves(<any>Promise.resolve(mockConnection));
+		sinon.stub(azdata.connection, 'getConnections').resolves(<any>Promise.resolve(getConnectionsResults));
+		sinon.stub(azdata.connection, 'listDatabases').resolves(['My Database']);
+
+		let schemaCompareResult = new SchemaCompareMainWindow(undefined, mockExtensionContext.object);
+		let dialog = new SchemaCompareDialogTest(schemaCompareResult, undefined, mockExtensionContext.object);
+
+		should.equal(dialog.getSourceDropdownValue(), undefined);
+		should.equal(dialog.getTargetDropdownValue(), undefined);
+
+		await dialog.openDialog();
+		await dialog.connectionButtonClick(false);
+
+		// Confirm source and target dropdowns have the new connection as its value
+		should.notEqual(dialog.getSourceDropdownValue(), undefined);
+		should((dialog.getSourceDropdownValue() as ConnectionDropdownValue).connection).deepEqual(mockConnectionProfile, `SourceDropdownValue: (Actual) ${(dialog.getSourceDropdownValue() as ConnectionDropdownValue).connection} (Expected) ${mockConnectionProfile}`);
+
+		should.notEqual(dialog.getTargetDropdownValue(), undefined);
+		should((dialog.getTargetDropdownValue() as ConnectionDropdownValue).connection).deepEqual(mockConnectionProfile, `TargetDropdownValue: (Actual) ${(dialog.getTargetDropdownValue() as ConnectionDropdownValue).connection} (Expected) ${mockConnectionProfile}`);
```

Comment:
it would be nice to also have a test to verify that if the target already has a selected connection, it doesn't change when a new connection is added for the source dropdown.

Reply:
Added this.